frontend-react/src/pages/admin/AdminRevHistory.tsx

```diff
@@ -0,0 +1,273 @@
+    return (
+        <Grid col={"fill"} className={"rs-maxwidth-vw80"}>
+            <Grid row gap="md" className={"rs-accord-list-row"}>
+                <Grid className={"rs-list-diffs-items"}>
+                    {!data ? (
+                        <div>{msg}</div>
+                    ) : (
+                        <Accordion
+                            bordered={false}
+                            items={dataToAccordionItems({
+                                key: "left",
+                                selectedKey: props.leftSelectedListItem,
+                                onClickHandler: props.onClickHandler,
+                                data,
+                            })}
+                        />
+                    )}
+                </Grid>
+                <Grid className={"rs-list-diffs-items"}>
+                    {!data ? null : (
+                        <Accordion
+                            bordered={false}
+                            items={dataToAccordionItems({
+                                key: "right",
+                                selectedKey: props.rightSelectedListItem,
+                                onClickHandler: props.onClickHandler,
+                                data,
+                            })}
+                        />
+                    )}
+                </Grid>
+            </Grid>
+        </Grid>
+    );
+};
+
+/** main page, not exported here because it should only be loaded via AdminRevHistoryWithAuth() **/
+const AdminRevHistory = () => {
+    const { org, settingType } = useParams<SettingRevisionParams>(); // props past to page via the route/url path args
+    const [leftJson, setLeftJson] = useState("");
+    const [rightJson, setRightJson] = useState("");
+    // used to highlight which item is selected.
+    const [leftSelectedListItem, setLeftSelectedListItem] = useState("");
+    const [rightSelectedListItem, setRightSelectedListItem] = useState("");
+    const [leftItem, setLeftItem] = useState<SettingRevision | null>(null);
+    const [rightItem, setRightItem] = useState<SettingRevision | null>(null);
+
+    const onClickHandler = useCallback(
+        (key: string, itemClickedKey: string, data: SettingRevision) => {
+            const normalizeJson = (jsonStr: string): string =>
+                JSON.stringify(JSON.parse(jsonStr), jsonSortReplacer, 2);
+            const prettyJson = normalizeJson(data.settingJson);
+
+            switch (key) {
+                case "left":
+                    setLeftSelectedListItem(itemClickedKey);
+                    setLeftJson(prettyJson);
+                    setLeftItem(data);
+                    break;
+                case "right":
+                    setRightSelectedListItem(itemClickedKey);
+                    setRightJson(prettyJson);
+                    setRightItem(data);
+                    break;
+            }
+        },
+        []
+    );
+
+    return (
+        <>
+            <Helmet>
```

Comment:
use <BasicHelmet>

Reply:
Great catch. This PR is taking so long, things are changing underneath it.